src/python/pants/backend/go/util_rules/third_party_pkg.py

```diff
@@ -327,23 +330,77 @@ async def _check_go_sum_has_not_changed(
             )
 
 
+@rule
+async def process_find_go_packages(
+    request: ModulePackageListRequest,
+    bash: BashBinary,
+    find: FindBinary,
+) -> ModulePackageList:
+    """This connects the (cached) go mod download steps with the package analyzer steps by finding
+    likely package directories within a module."""
+
+    # As background, several other options did not work for this layer:
+    # - `go list` provides the same package paths, but fails on modules which do not use go.sum
+    # - `os.walk` could process the named cache folder directly, but running a Process allows caching
+
+    find_script = FileContent(
+        "__run_find.sh",
+        textwrap.dedent(
+            f"""\
+            {find.path} {request.path} -type f -name '*.go'
+            """
+        ).encode("utf-8"),
+    )
+
+    digest = await Get(Digest, CreateDigest([find_script]))
+
+    list_pkg_result = await Get(
+        FallibleProcessResult,
+        Process(
+            # Use a named_cache symlink to reference downloaded go modules
+            append_only_caches=gosdk.sdk_cache(),
+            argv=[bash.path, find_script.path],
+            input_digest=digest,
+            description=f"find .go files for {request.path}",
+            level=LogLevel.DEBUG,
+        ),
+    )
+
+    all_dirs = set()
+    if list_pkg_result.exit_code != 0 or len(list_pkg_result.stdout) == 0:
+        logger.warning(f"failed to list {request.path}: {list_pkg_result.stderr.decode()}")
```

Comment:
Is this actually a failure or just no Go files found (in which case maybe the warning should state that Go files were expected)?

Reply:
I'm seeing the error cases come from invalid source path (zero go files found is fine).  So I'll switch to ProcessResult and let exit code errors bubble up.